homeassistant/components/squeezebox/sensor.py

```diff
@@ -0,0 +1,122 @@
+"""Platform for sensor integration for squeezebox."""
+
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import EntityCategory, UnitOfTime
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import SqueezeboxConfigEntry
+from .const import (
+    STATUS_SENSOR_INFO_TOTAL_ALBUMS,
+    STATUS_SENSOR_INFO_TOTAL_ARTISTS,
+    STATUS_SENSOR_INFO_TOTAL_DURATION,
+    STATUS_SENSOR_INFO_TOTAL_GENRES,
+    STATUS_SENSOR_INFO_TOTAL_SONGS,
+    STATUS_SENSOR_LASTSCAN,
+    STATUS_SENSOR_NEWPLUGINS,
+    STATUS_SENSOR_NEWVERSION,
+    STATUS_SENSOR_OTHER_PLAYER_COUNT,
+    STATUS_SENSOR_PLAYER_COUNT,
+)
+from .coordinator import LMSStatusDataUpdateCoordinator
+from .entity import LMSStatusEntity
+
+SENSORS: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key=STATUS_SENSOR_INFO_TOTAL_ALBUMS,
+        state_class=SensorStateClass.TOTAL,
+        icon="mdi:album",
+    ),
+    SensorEntityDescription(
+        key=STATUS_SENSOR_INFO_TOTAL_ARTISTS,
+        state_class=SensorStateClass.TOTAL,
+        icon="mdi:account-music",
+    ),
+    SensorEntityDescription(
+        key=STATUS_SENSOR_INFO_TOTAL_DURATION,
+        state_class=SensorStateClass.TOTAL,
+        device_class=SensorDeviceClass.DURATION,
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+    ),
+    SensorEntityDescription(
+        key=STATUS_SENSOR_INFO_TOTAL_GENRES,
+        state_class=SensorStateClass.TOTAL,
+        icon="mdi:drama-masks",
+    ),
+    SensorEntityDescription(
+        key=STATUS_SENSOR_INFO_TOTAL_SONGS,
+        state_class=SensorStateClass.TOTAL,
+        icon="mdi:file-music",
+    ),
+    SensorEntityDescription(
+        key=STATUS_SENSOR_LASTSCAN,
+        device_class=SensorDeviceClass.TIMESTAMP,
+    ),
+    SensorEntityDescription(
+        key=STATUS_SENSOR_NEWPLUGINS,
+        entity_registry_visible_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key=STATUS_SENSOR_NEWVERSION,
+        entity_registry_visible_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
```

Comment:
Why don't we make this an update entity?

Reply:
So the information provide by LMS is very simple and unstructured making it an update entity didn't really add any value

Reply:
Can you elaborate?

Reply:
Sure all you get is a simple unstructured string with some embedder html that changes dependant on install type. if its not present no upgrade.

Reply:
Sample text:-
`A new version of Logitech Media Server is available (8.5.2 - 0). <a href="updateinfo.html?installerFile=/var/lib/squeezeboxserver/cache/updates/logitechmediaserver_8.5.2_amd64.deb" target="update">Click here for further information</a>.`

Reply:
Yes but that doesn't answer the question why an update entity doesn't fit?

Reply:
@pssc, you likely know this already but, you can get the latest server version from https://lyrion.org/lms-server-repository/stable.xml, hhttps://lyrion.org/lms-server-repository/latest.xml and https://lyrion.org/lms-server-repository/dev.xml.  This of course just gives you the current version for each track and you'd have to check the currently selected track (actually not sure how to do that) and the current platform and version to know if an upgrade is required.  I guess that's what the server does itself to throw up the text that you're checking (and to already have downloaded the new version in the background) and so it's likely checking the presence of the text as a flag for "upgrade required" may well be the simplest.

Reply:
Yes and requires a separate poll, parse and all that jazz so I have taken what we have in server status and can make an update entity work